### DISCUSSION TITLE
SetValue of conditional Field

### DISCUSSION THREAD

**Discussion by j0hannr** [react-hook-form/react-hook-form#8131] [Q&A] [answered]
I have multiple conditional fields in my Form.
I want to set the values of all fields when the page is loaded, unfortunatly this does not work with conditional fields.

[CodeSandBox](https://codesandbox.io/s/react-hook-form-v7-setvalue-of-conditional-field-hotpyz)! - The Second Question is not Set on load

Thanks for any help! :)

**Comment by Moshyfawn** [comment, marked as answer]
To reset your form values on data load, use [the `reset` form method](https://react-hook-form.com/api/useform/reset) instead of `setValue`. [Here's a working example](https://codesandbox.io/s/react-hook-form-v7-setvalue-of-conditional-field-forked-punmdz?file=%2Fsrc%2Findex.js)

```diff
const {
- setValue
+ reset
} = useForm()

useEffect(() => {
-  setValue("roofType", fT);
-  setValue("roofInhabited", rI);
+  reset({
+    roofType: fT,
+    roofInhabited: rI
+  })
- }, [fT, rI, setValue])
+ }, [fT, rI, reset])
```

If the form's default values are available on page load, you can simply provide them as `defaultValues` prop to `useForm` hook
```js
useForm({
  defaultValues: {
    roofType: fT,
    roofInhabited: rI
  }
})
```

**Reply by j0hannr** [reply]
It works, thanks for the solution! :)
